File: app/src/main/java/com/lazylee/lzywanandroid/data/dao/HotKeyDao.java
package com.lazylee.lzywanandroid.data.dao;

import com.lazylee.lzywanandroid.data.entity.HotKey;

import java.util.List;

import androidx.room.Dao;
import androidx.room.Insert;
import androidx.room.OnConflictStrategy;
import androidx.room.Query;

@Dao
public interface HotKeyDao {

    @Query("Select * From HotKey")
    List<HotKey> getAll();

    @Insert(onConflict = OnConflictStrategy.REPLACE)
    void insertHotKeys(HotKey... hotKeys);

    @Insert(onConflict = OnConflictStrategy.REPLACE)
    void insertHotKey(HotKey hotKey);
}

File: app/src/main/java/com/lazylee/lzywanandroid/data/dao/HotKeyDao.kt
package com.lazylee.lzywanandroid.data.dao

import com.lazylee.lzywanandroid.data.entity.HotKey

import androidx.room.Dao
import androidx.room.Insert
import androidx.room.OnConflictStrategy
import androidx.room.Query

@Dao
interface HotKeyDao {

    @get:Query("Select * From HotKey")
    val all: List<HotKey>

    @Insert(onConflict = OnConflictStrategy.REPLACE)
    fun insertHotKeys(vararg hotKeys: HotKey)

    @Insert(onConflict = OnConflictStrategy.REPLACE)
    fun insertHotKey(hotKey: HotKey)
}
